src/Standards/Squiz/Sniffs/ControlStructures/ForLoopDeclarationSniff.php

```diff
@@ -76,7 +76,7 @@ public function process(File $phpcsFile, $stackPtr)
         $tokens = $phpcsFile->getTokens();
 
         $openingBracket = $phpcsFile->findNext(T_OPEN_PARENTHESIS, $stackPtr);
-        if ($openingBracket === false) {
+        if ($openingBracket === false || isset($tokens[$openingBracket]['parenthesis_closer']) === false) {
             $error = 'Possible parse error: no opening parenthesis for FOR keyword';
```

Comment:
I wonder if the error message should be changed as in some cases it will be displayed when the opening parenthesis is present and the closing parenthesis is missing?

Reply:
Good point. I didn't look too closely at the message as it will be removed in 4.0 anyway (per https://github.com/squizlabs/PHP_CodeSniffer/issues/2455), but yeah, might as well make it be correct until that time.

Note: I will not change the _error code_ though as that could be considered a breaking change.

Reply:
Thanks for updating the error message!